### PR TITLE
Allow setup different layouts per virtual desktop

### PR DESCRIPTION
**Pull request by sr-tream**
I have 2 virtual desktops: 1 - for browser, players, etc; 2 - for development.

For 1st virtual desktop, I want to use **Monocle Layout**, to open all windows maximized.
For 2nd virtual desktop, I want to use **Tile layout**, to open debuggable process side-by-side to IDE.

This PR solve this case.

### CODE REVIEW

File: res/config.ui [hunk 1]
```diff
@@ -23,7 +23,7 @@
    <item row="0" column="0">
     <widget class="QTabWidget" name="configTabs">
      <property name="currentIndex">
-      <number>4</number>
+      <number>3</number>
```
[the remark this answers is not in the document]
Comment (sr-tream) [reply res/config.ui:26]: QtDesugner automatically changes this value. I don't know what is it

File: src/engine/layoutstore.ts [hunk 2]
```diff
@@ -28,20 +28,24 @@ class LayoutStoreEntry {
...
-      let [cfg_output, cfg_screen_id_str] = entry.split(":");
+      let cfg = entry.split(":");
+      let cfg_output = cfg[0];
+      let cfg_desktop = cfg.length == 2 ? undefined : cfg[1];
+      let cfg_screen_id_str = cfg.length == 2 ? cfg[1] : cfg[2];
       let cfg_screen_id = parseInt(cfg_screen_id_str);
 
       if (
-        output_name === cfg_output &&
+        (output_name === cfg_output || cfg_output === '') &&
```
[the remark this answers is not in the document]
Comment (sr-tream) [reply src/engine/layoutstore.ts:47]: It's for any screen on a specific desktop.

E.g. `:Work desktop:3` - enable layout 3 for each screen when user on virtual desktop "Work desktop"

File: res/config.ui (hunk 1 above)
[the remark this answers is not in the document]
Comment (sr-tream) [reply res/config.ui:26]: Oh its a default tab, wait I fixing it

File: src/engine/layoutstore.ts [hunk 3]
```diff
@@ -133,7 +137,8 @@ class LayoutStore {
         delete this.store[key_without_activity];
       } else {
         let output_name = key.slice(0, key.indexOf("@"));
-        this.store[key] = new LayoutStoreEntry(output_name);
+        let desktop_name = i2 !== -1 ? key.slice(i2 + 1) : undefined;
```
Comment (anametologin) [inline src/engine/layoutstore.ts:140]: cannot be undefined. desktop_name as split of HDMI-A-1@f381c9cf-cb90-4ade-8b3f-24ae0002d366#Desktop 1

File: src/engine/layoutstore.ts [hunk 4]
```diff
@@ -28,20 +28,24 @@ class LayoutStoreEntry {
   private layouts: { [key: string]: ILayout };
   private previousID: string;
 
-  constructor(output_name: string) {
+  constructor(output_name: string, desktop_name?: string) {
```
Comment (anametologin) [inline src/engine/layoutstore.ts:31]: desktop_name always exists

**Comment by anametologin** [comment]
:2 - makes layout 2  on all screens
:Desktop 1:2 - makes layout 2 on all desktops with name Desktop 1
I will write it in readme...

File: src/engine/layoutstore.ts (hunk 3 above)
Comment (sr-tream) [reply src/engine/layoutstore.ts:140]: Name generated in [KWinSurface.generateId](https://github.com/anametologin/krohnkite/blob/master/src/driver/kwin/kwinsurface.ts#L22) and `#Desktop 1` part may be missing.

It writes as surface ID [here](https://github.com/anametologin/krohnkite/blob/master/src/driver/kwin/kwinsurface.ts#L49) and passed to [LayoutStore.getEntry](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L123) ([1](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L110), [2](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L115), [3](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L120)).

File: src/engine/layoutstore.ts (hunk 4 above)
Comment (sr-tream) [reply src/engine/layoutstore.ts:31]: Depended on "Use separate layouts for each Desktop" setting

**Pull request merged**